### PR TITLE
ocrvs-2774 Amend Question label, placeholder, description & tool tip for localisation.

### PR DESCRIPTION
**Pull request by sadmananik**

### CODE REVIEW

File: packages/config/src/handlers/updateQuestion/handler.ts [hunk 1]
```diff
@@ -53,17 +53,23 @@ export default async function updateQuestion(
...
-  id: Joi.string().required(),
-  defaultMessage: Joi.string(),
-  description: Joi.string()
+const messageSchema = Joi.array().items({
+  lang: Joi.string(),
+  descriptor: Joi.object({
+    id: Joi.string().required(),
+    defaultMessage: Joi.string(),
+    description: Joi.string()
+  })
```
Comment (Zangetsu101) [inline packages/config/src/handlers/updateQuestion/handler.ts:62]: This is being used in separate places. Maybe we can use a common one?

File: packages/gateway/src/features/formDraft/schema.graphql [hunk 2]
```diff
@@ -33,11 +33,19 @@ input MesssageDescriptorInput {
...
 input QuestionInput {
   id: ID
   fieldId: String!
-  label: MesssageDescriptorInput
-  placeholder: MesssageDescriptorInput
+  label: [MesssageInput]
+  placeholder: [MesssageInput]
+  description: [MesssageInput]
+  tooltip: [MesssageInput]
+  errorMessage: [MesssageInput]
```
Comment (Zangetsu101) [inline packages/gateway/src/features/formDraft/schema.graphql:48]:
```suggestion
  label: [MesssageInput!]
  placeholder: [MesssageInput!]
  description: [MesssageInput!]
  tooltip: [MesssageInput!]
  errorMessage: [MesssageInput!]
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]
The graphql types will need to be handled correctly so that the fields which are **compulsory** will be **compulsory** and not **optional**

File: packages/client/src/forms/configuration/customUtils.ts [hunk 3]
```diff
@@ -74,6 +75,14 @@ export function createCustomGroup(
   customQuestionConfigurations.push(newCustomGroup)
 }
 
+function getDefaultLanguageMessage(messages: IMessage[] | undefined) {
+  const languages = getDefaultLanguage()
```
[the remark this answers is not in the document]
Comment (sadmananik) [reply packages/client/src/forms/configuration/customUtils.ts:79]: Changed

File: packages/config/src/handlers/updateQuestion/handler.ts (hunk 1 above)
Comment (sadmananik) [reply packages/config/src/handlers/updateQuestion/handler.ts:62]: Yeah. I have to import it from createQuestions. Thanks.

File: packages/gateway/src/features/formDraft/schema.graphql (hunk 2 above)
Comment (sadmananik) [reply packages/gateway/src/features/formDraft/schema.graphql:48]:
```
type Messsage {
  lang: String!
  descriptor: MesssageDescriptor!
}

type MesssageDescriptor {
  id: String!
  description: String!
  defaultMessage: String!
}
```

File: packages/gateway/src/features/questions/schema.graphql [hunk 4]
```diff
@@ -20,11 +20,24 @@ type MesssageDescriptor {
...
 type Question {
   _id: ID!
   fieldId: String!
-  label: MesssageDescriptor
-  placeholder: MesssageDescriptor
+  label: [Messsage]
+  placeholder: [Messsage]
+  description: [Messsage]
+  tooltip: [Messsage]
+  errorMessage: [Messsage]
```
[the remark this answers is not in the document]
Comment (sadmananik) [reply packages/gateway/src/features/questions/schema.graphql:40]:
```
type MesssageDescriptor {
  id: String!
  description: String!
  defaultMessage: String!
}
```

File: packages/gateway/src/graphql/schema.graphql [hunk 5]
```diff
@@ -1,1512 +0,0 @@
...
-
-enum PaymentType {
-  MANUAL
-}
-
-enum PaymentOutcomeType {
-  COMPLETED
-  ERROR
-  PARTIAL
-}
```
Comment (Zangetsu101) [inline packages/gateway/src/graphql/schema.graphql:1496]: That's lots of lines being deleted. What caused it to change so much?

Comment (sadmananik) [reply packages/gateway/src/graphql/schema.graphql]: There was a problem with graphQL schema generation. Fixed now.

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]
Otherwise everything looks good.

File: packages/gateway/src/features/questions/schema.graphql [hunk 6]
```diff
@@ -38,8 +51,13 @@ type Question {
 input QuestionInput {
   id: ID
   fieldId: String!
-  label: MesssageDescriptorInput
-  placeholder: MesssageDescriptorInput
+  label: [MesssageInput!]
+  placeholder: [MesssageInput!]
+  label: [MesssageInput!]
+  placeholder: [MesssageInput!]
```
[the remark this answers is not in the document]
Comment (sadmananik) [reply packages/gateway/src/features/questions/schema.graphql]: Fixed

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**